### PR TITLE
Add tests for time travel

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/2493.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217424217)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 7e76e5604e727216cf8daaf6959f43ac76eecf20
Using browser at version BuildID 20170331102157; SourceStamp 8df9fabf2587b7020889755acb9e75b664fe13cf
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5311/html/webappapis/timers/negative-timeout.html">/html/webappapis/timers/negative-timeout.html</a></summary>


|                                         Subtest                                          | Results |     Messages     |
|------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                          | ERROR   |                  |
| `setTimeout with negative arguments travels into the past, affecting Date`               | TIMEOUT | `Test timed out` |
| `setTimeout with negative arguments travels into the past, affecting API implementation` | TIMEOUT | `Test timed out` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217424218)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 7e76e5604e727216cf8daaf6959f43ac76eecf20
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5311/html/webappapis/timers/negative-timeout.html">/html/webappapis/timers/negative-timeout.html</a></summary>


|                                         Subtest                                          | Results |     Messages     |
|------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                          | ERROR   |                  |
| `setTimeout with negative arguments travels into the past, affecting Date`               | TIMEOUT | `Test timed out` |
| `setTimeout with negative arguments travels into the past, affecting API implementation` | TIMEOUT | `Test timed out` |

</details>
</details>

**Comment by foolip** [comment]
I just merged the spec change, but now I'm thinking there are more edge cases to be tested here. What happens if someone uses this API to change the HTML namespace to http://www.w3.org/1998/xhtml, will that cause all of our parser tests to fail? Seems like we might get some tricky timeline merge conflicts?

**Comment by foolip** [comment]
I see the tests are timing out, but that's probably a testharness.js limitation, as its timeout is positive.

**Comment by domenic** [comment]
> I just merged the spec change, but now I'm thinking there are more edge cases to be tested here. What happens if someone uses this API to change the HTML namespace to http://www.w3.org/1998/xhtml, will that cause all of our parser tests to fail? Seems like we might get some tricky timeline merge conflicts?

Hmm, I can see the worry. Perhaps we could use the API to go back and get rid of XML namespaces entirely? Maybe even XML itself. I think that would be the best solution to the problem.

**Comment by mathiasbynens** [comment]
Tests for `setInterval` are missing.

**Comment by mathiasbynens** [comment]
Fixed by running `setTimeout(addIntervalTests, 1 * 1000 * 60 * 60);` in Chrome Canary with `chrome://flags/#enable-experimental-time-travel` enabled.

File: html/webappapis/timers/negative-timeout.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    assert_less_than(yesterday, today);
+  }, -24 * 60 * 60 * 1000);
+}, "setTimeout with negative arguments travels into the past, affecting Date");
+
+async_test(t => {
+  // Nobody implemented service workers before 2015
+  const travelTo = new Date(2015, 0, 1);
+  const travelTime = travelTo - Date.now();
+
+  // Be sure not to use modern JS features like arrow functions or const; they don't work in the past
```
Comment (mathiasbynens) [inline html/webappapis/timers/negative-timeout.html]: This changes everything! @hzoo, heads-up: babel-env should give up and transpile everything when it detects `setTimeout`/`setInterval` usage.

**Comment by foolip** [comment]
Cursed implementers didn't implement this at any point in the future or I'm sure we would have known about it. Ah well, closing this.

**Pull request closed without merging**